pkg/networkmanager/interfaces.js

```diff
@@ -2395,14 +2395,11 @@ PageNetworkInterface.prototype = {
         var self = this;
 
         function delete_connection_and_slaves(con) {
-            return cockpit.all(con.delete_(),
-                               cockpit.all(con.Slaves.map(function (s) {
-                                   return free_slave_connection(s);
-                               })));
+            return Promise.all([con.delete_(), con.slaves.map(s => free_slave_connection(s))]);
```

Comment:
While you are at simplifying source code, why not go all the way and use `map(free_slave_connection)` :-)

Reply:
Of course :)